src/test/java/co/rsk/federate/FederatorSupportTest.java

```diff
@@ -186,6 +191,139 @@ void getProposedFederation_whenAddressStringIsValid_shouldReturnAddress() {
         assertEquals(DEFAULT_ADDRESS.toString(), result.get().toString());
     }
 
+    @Test
+    void getProposedFederationSize_whenSizeIsNull_shouldReturnEmptyOptional() {
+        // Arrange
+        when(bridgeTransactionSender.callTx(any(), eq(Bridge.GET_PROPOSED_FEDERATION_SIZE)))
+            .thenReturn(null);
+
+        // Act
+        Optional<Integer> result = federatorSupport.getProposedFederationSize();
+
+        // Assert
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    void getProposedFederationSize_whenSizeIsPresent_shouldReturnInteger() {
+        // Arrange
+        BigInteger expectedSize = BigInteger.valueOf(9);
+        when(bridgeTransactionSender.callTx(any(), eq(Bridge.GET_PROPOSED_FEDERATION_SIZE)))
+            .thenReturn(expectedSize);
+
+        // Act
+        Optional<Integer> result = federatorSupport.getProposedFederationSize();
+
+        // Assert
+        assertTrue(result.isPresent());
+        assertEquals(expectedSize.intValue(), result.get());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenPublicKeyBytesIsNull_shouldReturnEmptyOptional() {
+        // Arrange
+        int index = 0;
+        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
+        when(bridgeTransactionSender.callTx(
+               any(),
+               eq(Bridge.GET_PROPOSED_FEDERATOR_PUBLIC_KEY_OF_TYPE),
+               eq(new Object[]{ index, keyType.getValue() })))
+            .thenReturn(null);
+
+        // Act
+        Optional<ECKey> result = federatorSupport.getProposedFederatorPublicKeyOfType(index, keyType);
+
+        // Assert
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenPublicKeyBytesArePresent_shouldReturnECKey() {
+        // Arrange
+        int index = 0; 
+        FederationMember.KeyType keyType = FederationMember.KeyType.BTC;
+        ECKey expectedKey = ECKey.fromPublicOnly(PUBLIC_KEY);
+        when(bridgeTransactionSender.callTx(
+               any(),
+               eq(Bridge.GET_PROPOSED_FEDERATOR_PUBLIC_KEY_OF_TYPE),
+               eq(new Object[]{ index, keyType.getValue() })))
+            .thenReturn(PUBLIC_KEY);
+
+        // Act
+        Optional<ECKey> result = federatorSupport.getProposedFederatorPublicKeyOfType(index, keyType);
+
+        // Assert
+        assertTrue(result.isPresent());
+        assertArrayEquals(expectedKey.getPubKey(), result.get().getPubKey());
+    }
+
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenKeyTypeIsNull_shouldThrowNullPointerException() {
+        // Arrange
+        int index = 0;
+        
+        // Act & Assert
+        assertThrows(NullPointerException.class, () -> {
```

Comment:
is this what we expect? a npe?

Reply:
i mean, is it the best thing to do?

Reply:
Yes, since we are requiring it to be non null it will throw an NPE. Check: https://docs.oracle.com/javase/10/docs/api/java/util/Objects.html#requireNonNull(T)

Reply:
you right, thanks!